daemon/daemon_unix.go

```diff
@@ -375,6 +390,13 @@ func verifyPlatformContainerSettings(daemon *Daemon, hostConfig *containertypes.
 		warnings = append(warnings, "IPv4 forwarding is disabled. Networking will not work.")
 		logrus.Warnf("IPv4 forwarding is disabled. Networking will not work")
 	}
+	if hostConfig.CgroupParent != "" && daemon.usingSystemd() {
+		// CgroupParent for systemd cgroup should be named as "xxx.slice"
+		length := len(hostConfig.CgroupParent)
+		if length <= 6 || hostConfig.CgroupParent[length-6:] != ".slice" {
```

Comment:
> The whole point of being able to set cgroup-parent is so you can put a container in a cgroup where you control the resource limitations (like a slice in systemd, or just a cgroup that you're managing).

The point is you can put a container in a cgroup which you care about, it's common that not every user cares about all subsystems, so if the slice name you assigned doesn't not exist in some subsystems, you don't care the resource limitations for these subsystems, by default it'll be unlimited.

If we restrict rule like `--cgroup-parent=/a.slice/b.slice`, and demand this `/a.slice/b.slice` has to be existed in all subsystems, it'll be too rigorous and makes this unusable.